werkzeug/wrappers.py

```diff
@@ -1443,12 +1522,21 @@ def make_conditional(self, request_or_environ):
             # wsgiref.
             if 'date' not in self.headers:
                 self.headers['Date'] = http_date()
+            accept_ranges = _clean_accept_ranges(accept_ranges)
+            if not self._process_range_request(environ, complete_length, accept_ranges):
+                from werkzeug.exceptions import RequestedRangeNotSatisfiable
+                raise RequestedRangeNotSatisfiable(complete_length)
             if self.automatically_set_content_length and 'content-length' not in self.headers:
                 length = self.calculate_content_length()
                 if length is not None:
                     self.headers['Content-Length'] = length
-            if not is_resource_modified(environ, self.headers.get('etag'), None,
-                                        self.headers.get('last-modified')):
+            if (
+                self.status_code != 206
+                and not is_resource_modified(
+                    environ, self.headers.get('etag'),
+                    None, self.headers.get('last-modified')
```

Comment:
Ah I see, sorry about that. I think I've found a way to restore this entire conditional to its original state:

    if not self._process_range_request(...):
        ...
    elif not is_resource_modified(...):
        ...

    if self.automatically_set_content_length ...:
        ...
    return self